lib/imap.c

```diff
@@ -692,12 +698,12 @@ static CURLcode imap_perform_fetch(struct connectdata *conn)
 
   /* Send the FETCH command */
   if(imap->partial)
-    result = imap_sendf(conn, "FETCH %s BODY[%s]<%s>",
+    result = imap_sendf(conn, "UID FETCH %s BODY[%s]<%s>",
                         imap->uid,
                         imap->section ? imap->section : "",
                         imap->partial);
   else
-    result = imap_sendf(conn, "FETCH %s BODY[%s]",
+    result = imap_sendf(conn, "UID FETCH %s BODY[%s]",
```

Comment:
So we should *never* do plain `FETCH` (without `UID` in front of it) with imap anymore? Can't a URL be specified that uses that?

Reply:
By RFC 5092  it is, except using CUSTOMREQUEST FETCH in plain mode.

Reply:
Url SEARCH scheme return message sequence number, FETCH scheme return unique identifier. Indeed, this is uncoordinated.

Could we make a little extension? That is, url FETCH scheme should comply with RFC , but when uid number is zero-prefixed, like 0123, then we treat it as sequence number. Thus, we coordinate the two schemes.

Reply:
> when uid number is zero-prefixed, like 0123, then we treat it as sequence number

That feels like very weird and unexpected. I don't think that would help much.